src/fhirdefs/FHIRDefinitions.ts

```diff
@@ -1,35 +1,19 @@
+import { cloneDeep, flatten } from 'lodash';
+import { FHIRDefinitions as BaseFHIRDefinitions } from 'fhir-package-loader';
 import { Type, Metadata, Fishable } from '../utils';
 import { IMPLIED_EXTENSION_REGEX, materializeImpliedExtension } from './impliedExtensions';
-import { cloneDeep, flatten } from 'lodash';
 import { STRUCTURE_DEFINITION_R4_BASE } from '../fhirtypes';
 
-export class FHIRDefinitions implements Fishable {
-  private resources: Map<string, any>;
-  private logicals: Map<string, any>;
-  private profiles: Map<string, any>;
-  private extensions: Map<string, any>;
-  private types: Map<string, any>;
-  private valueSets: Map<string, any>;
-  private codeSystems: Map<string, any>;
-  private implementationGuides: Map<string, any>;
+export class FHIRDefinitions extends BaseFHIRDefinitions implements Fishable {
   private predefinedResources: Map<string, any>;
   private supplementalFHIRDefinitions: Map<string, FHIRDefinitions>;
-  private packageJsons: Map<string, any>;
   packages: string[];
```

Comment:
Hmm this property gave me a lot of trouble, and I thought it was necessary still because the way the `FHIRDefinitions` work between SUSHI and FPL are slightly different, and I thought this property was the key to that difference. But you're right. It isn't adding to that list anymore, and I thought it was. So I'm going to look more into this and see if it actually should be using this property. If not, I'll just update the tests to get rid of it there as well.

Reply:
I do think it could potentially be useful to have a property that just lets you know which packages are contained in a `FHIRDefinitions` instance. Maybe we can add the `packages` array to the base `FHIRDefinitions` class over on the FPL side, and then add a getter that returns the current `package` property and the package property of all the `childDefs` in an array 🤔. I'm not sure if the need for that would ever come up though.

Reply:
Phew 😅😰 I made it.

Turns out `packages` was really only used in tests. I spent a bit trying to rework how SUSHI actually uses the FHIRDefinitions class as part of looking in to this because it would probably be a little cleaner and let us get cleaner access to the list of packages that it updates. However, it really wasn't working completely correctly, probably would have involved more reworking than we really need right now, and would impact a bunch of tests.

So instead I just kept track of an array in the tests as each package is "loaded" and check against that array, rather than depend on a property on FHIRDefinitions. That's all done in [6f38ef1](https://github.com/FHIR/sushi/pull/1026/commits/6f38ef12511fedb585b2aa6d2125d9404955a6ea).